### ISSUE TITLE
stemless note interaction with tie directions

### ISSUE THREAD

**Issue by craigsapp** [rism-ch/verovio#534]
When the starting note of a tie is stemless, the tie orientation algorithm treats it as a stem-up note, and the tie orientation is assigned only from the starting notes stem directory (if on a single-layer staff).

<img width="495" alt="screen shot 2017-04-11 at 11 42 46 am" src="https://cloud.githubusercontent.com/assets/3487289/24925401/fab6a31a-1eac-11e7-9d56-9a441b31e3b6.png">

It would be useful to change the algorithm to either:

(1) if the start note is stemless then check if the end note has a stem, and use that stem direction to decide on an orientation for the tie.

or

(2) assign a stem direction to the stemless note (i.e., check its position on the staff

The second case seems to be in effect, but does not work for the middle line on the staff.  Here is moving down a diatonic step:

<img width="501" alt="screen shot 2017-04-11 at 11 51 41 am" src="https://cloud.githubusercontent.com/assets/3487289/24925494/50400984-1ead-11e7-89cf-e1f485934762.png">

Or up a step:

<img width="506" alt="screen shot 2017-04-11 at 11 52 43 am" src="https://cloud.githubusercontent.com/assets/3487289/24925527/67a9be1c-1ead-11e7-8360-9edbc3100c5d.png">

Test data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2017-04-11T11:42:22" version="0.9.14-dev-c2f0a6e-dirty">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
        <workDesc>
            <work>
                <titleStmt>
                    <title />
                </titleStmt>
            </work>
        </workDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-0000000825138343">
                        <staffGrp xml:id="staffgrp-0000001603816504">
                            <staffDef xml:id="staffdef-0000000944139027" n="1" clef.shape="G" clef.line="2" meter.count="4" meter.unit="4" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-0000001122431159">
                        <measure xml:id="measure-L3" n="1" type="m-1">
                            <staff xml:id="staff-L3F1N1" n="1">
                                <layer xml:id="layer-L3F1N1" n="1">
                                    <note xml:id="note-L4F1"oct="4" pname="b">
                                        <accid xml:id="accid-L4F1" accid.ges="n" />
                                    </note>
                                </layer>
                            </staff>
                            <tie xml:id="tie-L4F1-L6F1" startid="#note-L4F1" endid="#note-L6F1" />
                        </measure>
                        <measure xml:id="measure-L5" type="m--1">
                            <staff xml:id="staff-L5F1N1" n="1">
                                <layer xml:id="layer-L5F1N1" n="1">
                                    <note xml:id="note-L6F1" oct="4" pname="b">
                                        <accid xml:id="accid-L6F1" accid.ges="n" />
                                    </note>
                                    <note xml:id="note-L7F1" oct="4" pname="b">
                                        <accid xml:id="accid-L7F1" accid.ges="n" />
                                    </note>
                                </layer>
                            </staff>
                            <tie xml:id="tie-L7F1-L9F1" startid="#note-L7F1" endid="#note-L9F1" />
                        </measure>
                        <measure xml:id="measure-L8" n="3" right="end" type="m-3">
                            <staff xml:id="staff-L8F1N1" n="1">
                                <layer xml:id="layer-L8F1N1" n="1">
                                    <note xml:id="note-L9F1" oct="4" pname="b">
                                        <accid xml:id="accid-L9F1" accid.ges="n" />
                                    </note>
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by craigsapp** [comment]
The behavior also affects tied longs:

<img width="849" alt="screen shot 2017-04-13 at 7 47 57 pm" src="https://cloud.githubusercontent.com/assets/3487289/25031521/5298c500-2082-11e7-93fc-402758b58406.png">

Longs don't seem to have stems which might be related to the problem.

Test MEI data:

```
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2017-04-13T19:47:42" version="0.9.14-dev-c2f0a6e-dirty">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
        <workDesc>
            <work>
                <titleStmt>
                    <title />
                </titleStmt>
            </work>
        </workDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-0000001549918363">
                        <staffGrp xml:id="staffgrp-0000000195082175">
                            <staffDef xml:id="staffdef-0000000838892216" n="1" clef.shape="G" clef.line="2" meter.count="12" meter.unit="2" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-0000000975226836">
                        <measure xml:id="measure-L3" n="1" type="m-1">
                            <staff xml:id="staff-L3F1N1" n="1">
                                <layer xml:id="layer-L3F1N1" n="1">
                                    <note xml:id="note-L4F1" type="qon-0 qoff-24 b40-191 tie-start" dots="1" dur="long" oct="4" pname="a">
                                        <accid xml:id="accid-L4F1" accid.ges="n" />
                                    </note>
                                </layer>
                            </staff>
                            <tie xml:id="tie-L4F1-L6F1" startid="#note-L4F1" endid="#note-L6F1" />
                        </measure>
                        <measure xml:id="measure-L5" type="m--1">
                            <staff xml:id="staff-L5F1N1" n="1">
                                <layer xml:id="layer-L5F1N1" n="1">
                                    <note xml:id="note-L6F1" type="qon-24 qoff-48 b40-191 tie-stop" dots="1" dur="long" oct="4" pname="a">
                                        <accid xml:id="accid-L6F1" accid.ges="n" />
                                    </note>
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L7" type="m--1">
                            <staff xml:id="staff-L7F1N1" n="1">
                                <layer xml:id="layer-L7F1N1" n="1">
                                    <note xml:id="note-L8F1" type="qon-48 qoff-72 b40-197 tie-start" dots="1" dur="long" oct="4" pname="b">
                                        <accid xml:id="accid-L8F1" accid.ges="n" />
                                    </note>
                                </layer>
                            </staff>
                            <tie xml:id="tie-L8F1-L10F1" startid="#note-L8F1" endid="#note-L10F1" />
                        </measure>
                        <measure xml:id="measure-L9" type="m--1">
                            <staff xml:id="staff-L9F1N1" n="1">
                                <layer xml:id="layer-L9F1N1" n="1">
                                    <note xml:id="note-L10F1" type="qon-72 qoff-96 b40-197 tie-stop" dots="1" dur="long" oct="4" pname="b">
                                        <accid xml:id="accid-L10F1" accid.ges="n" />
                                    </note>
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L11" type="m--1">
                            <staff xml:id="staff-L11F1N1" n="1">
                                <layer xml:id="layer-L11F1N1" n="1">
                                    <note xml:id="note-L12F1" type="qon-96 qoff-120 b40-202 tie-start" dots="1" dur="long" oct="5" pname="c">
                                        <accid xml:id="accid-L12F1" accid.ges="n" />
                                    </note>
                                </layer>
                            </staff>
                            <tie xml:id="tie-L12F1-L14F1" startid="#note-L12F1" endid="#note-L14F1" />
                        </measure>
                        <measure xml:id="measure-L13" right="end" type="m--1">
                            <staff xml:id="staff-L13F1N1" n="1">
                                <layer xml:id="layer-L13F1N1" n="1">
                                    <note xml:id="note-L14F1" type="qon-120 qoff-144 b40-202 tie-stop" dots="1" dur="long" oct="5" pname="c">
                                        <accid xml:id="accid-L14F1" accid.ges="n" />
                                    </note>
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by craigsapp** [comment]
And by extension the same thing happens with maxima:

<img width="920" alt="screen shot 2017-04-13 at 8 02 29 pm" src="https://cloud.githubusercontent.com/assets/3487289/25031794/36725f7e-2084-11e7-9bde-7b3992391d16.png">

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2017-04-13T20:02:20" version="0.9.14-dev-c2f0a6e-dirty">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
        <workDesc>
            <work>
                <titleStmt>
                    <title />
                </titleStmt>
            </work>
        </workDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-0000000747371077">
                        <staffGrp xml:id="staffgrp-0000002079849440">
                            <staffDef xml:id="staffdef-0000001323769876" n="1" clef.shape="G" clef.line="2" meter.count="8" meter.unit="1" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-0000001887432280">
                        <measure xml:id="measure-L3" n="1" type="m-1">
                            <staff xml:id="staff-L3F1N1" n="1">
                                <layer xml:id="layer-L3F1N1" n="1">
                                    <note xml:id="note-L4F1" type="qon-0 qoff-32 b40-191 tie-start" dur="maxima" oct="4" pname="a">
                                        <accid xml:id="accid-L4F1" accid.ges="n" />
                                    </note>
                                    <note xml:id="note-L5F1" type="qon-32 qoff-64 b40-191 tie-stop" dur="maxima" oct="4" pname="a">
                                        <accid xml:id="accid-L5F1" accid.ges="n" />
                                    </note>
                                </layer>
                            </staff>
                            <tie xml:id="tie-L4F1-L5F1" startid="#note-L4F1" endid="#note-L5F1" />
                        </measure>
                        <measure xml:id="measure-L6" n="2" type="m-2">
                            <staff xml:id="staff-L6F1N1" n="1">
                                <layer xml:id="layer-L6F1N1" n="1">
                                    <note xml:id="note-L7F1" type="qon-64 qoff-96 b40-197 tie-start" dur="maxima" oct="4" pname="b">
                                        <accid xml:id="accid-L7F1" accid.ges="n" />
                                    </note>
                                    <note xml:id="note-L8F1" type="qon-96 qoff-128 b40-197 tie-stop" dur="maxima" oct="4" pname="b">
                                        <accid xml:id="accid-L8F1" accid.ges="n" />
                                    </note>
                                </layer>
                            </staff>
                            <tie xml:id="tie-L7F1-L8F1" startid="#note-L7F1" endid="#note-L8F1" />
                        </measure>
                        <measure xml:id="measure-L9" n="3" right="end" type="m-3">
                            <staff xml:id="staff-L9F1N1" n="1">
                                <layer xml:id="layer-L9F1N1" n="1">
                                    <note xml:id="note-L10F1" type="qon-128 qoff-160 b40-202 tie-start" dur="maxima" oct="5" pname="c">
                                        <accid xml:id="accid-L10F1" accid.ges="n" />
                                    </note>
                                    <note xml:id="note-L11F1" type="qon-160 qoff-192 b40-202 tie-stop" dur="maxima" oct="5" pname="c">
                                        <accid xml:id="accid-L11F1" accid.ges="n" />
                                    </note>
                                </layer>
                            </staff>
                            <tie xml:id="tie-L10F1-L11F1" startid="#note-L10F1" endid="#note-L11F1" />
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by craigsapp** [comment]
Another example from the Hummel trumpet concerto, where I added tied half notes at the start of the tie group (that are oriented correctly):

<img width="1019" alt="screen shot 2017-10-21 at 8 42 16 pm" src="https://user-images.githubusercontent.com/3487289/31857967-1580edb8-b6a1-11e7-8467-fe767c6162c8.png">

```
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title>Trumpet Concerto in E major</title>
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2017-10-21T20:42:33" version="2.0.0-dev-ae70647-dirty">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
        <workDesc>
            <work>
                <titleStmt>
                    <title xml:id="title-L2" analog="humdrum:OTL" type="main">Trumpet Concerto in E major</title>
                    <respStmt>
                        <persName xml:id="persname-L1" analog="humdrum:COM" role="composer">Johann Nepomuk Hummel</persName>
                    </respStmt>
                </titleStmt>
            </work>
        </workDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-0000001323478639" midi.bpm="132">
                        <staffGrp xml:id="staffgrp-0000002038703198">
                            <staffDef xml:id="staffdef-0000001750042188" clef.shape="G" clef.line="2" key.sig="4s" meter.count="4" meter.unit="4" n="1" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-0000000481051104">
                        <measure xml:id="measure-L3" n="0">
                            <staff xml:id="staff-0000001469113257" n="1">
                                <layer xml:id="layer-L3F1N1" n="1">
                                    <note xml:id="note-L10F1" dur="2" oct="4" pname="b" accid.ges="n" />
                                    <note xml:id="note-L11F1" dur="2" oct="4" pname="b" accid.ges="n" />
                                </layer>
                            </staff>
                            <dynam xml:id="dynam-L10F2" staff="1" tstamp="1.000000">pp</dynam>
                            <tie xml:id="tie-L10F1-L11F1" startid="#note-L10F1" endid="#note-L11F1" />
                            <tie xml:id="tie-L11F1-L13F1" startid="#note-L11F1" endid="#note-L13F1" />
                        </measure>
                        <measure xml:id="measure-L12" n="2">
                            <staff xml:id="staff-L12F1N1" n="1">
                                <layer xml:id="layer-L12F1N1" n="1">
                                    <note xml:id="note-L13F1" dur="1" oct="4" pname="b" accid.ges="n" />
                                </layer>
                            </staff>
                            <tie xml:id="tie-L13F1-L16F1" startid="#note-L13F1" endid="#note-L16F1" />
                        </measure>
                        <measure xml:id="measure-L14" n="3">
                            <staff xml:id="staff-L14F1N1" n="1">
                                <layer xml:id="layer-L14F1N1" n="1">
                                    <note xml:id="note-L16F1" dur="1" oct="4" pname="b" accid.ges="n" />
                                </layer>
                            </staff>
                            <dir xml:id="dir-L16F1" staff="1" tstamp="1.000000">cresc.</dir>
                            <tie xml:id="tie-L16F1-L18F1" startid="#note-L16F1" endid="#note-L18F1" />
                        </measure>
                        <measure xml:id="measure-L17" n="4">
                            <staff xml:id="staff-L17F1N1" n="1">
                                <layer xml:id="layer-L17F1N1" n="1">
                                    <note xml:id="note-L18F1" dots="1" dur="2" oct="4" pname="b" stem.dir="down" accid.ges="n" />
                                    <note xml:id="note-L19F1" dur="4" oct="5" pname="d" stem.dir="down" accid.ges="s" />
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L20" n="5">
                            <staff xml:id="staff-L20F1N1" n="1">
                                <layer xml:id="layer-L20F1N1" n="1">
                                    <note xml:id="note-L21F1" dur="1" oct="5" pname="f" accid.ges="s" />
                                </layer>
                            </staff>
                            <dynam xml:id="dynam-L21F2" staff="1" tstamp="1.000000">f</dynam>
                            <tie xml:id="tie-L21F1-L23F1" startid="#note-L21F1" endid="#note-L23F1" />
                        </measure>
                        <measure xml:id="measure-L22" n="6">
                            <staff xml:id="staff-L22F1N1" n="1">
                                <layer xml:id="layer-L22F1N1" n="1">
                                    <note xml:id="note-L23F1" dur="1" oct="5" pname="f" accid.ges="s" />
                                </layer>
                            </staff>
                            <tie xml:id="tie-L23F1-L25F1" startid="#note-L23F1" endid="#note-L25F1" />
                        </measure>
                        <measure xml:id="measure-L24" n="7">
                            <staff xml:id="staff-L24F1N1" n="1">
                                <layer xml:id="layer-L24F1N1" n="1">
                                    <note xml:id="note-L25F1" dur="1" oct="5" pname="f" accid.ges="s" />
                                </layer>
                            </staff>
                            <tie xml:id="tie-L25F1-L27F1" startid="#note-L25F1" endid="#note-L27F1" />
                        </measure>
                        <measure xml:id="measure-L26" n="8">
                            <staff xml:id="staff-L26F1N1" n="1">
                                <layer xml:id="layer-L26F1N1" n="1">
                                    <note xml:id="note-L27F1" dots="1" dur="2" oct="5" pname="f" stem.dir="down" accid.ges="s" />
                                    <rest xml:id="rest-L28F1" dur="4" />
                                </layer>
                            </staff>
                            <fermata xml:id="fermata-L27F1" staff="1" startid="#note-L27F1" place="above" />
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by lpugin** [comment]
Fixed with a33469de7456c329176c8e2925bb6e12df61199c
